controllers/storagecluster/backingstorageclasses.go

```diff
@@ -0,0 +1,162 @@
+	storageClass.ObjectMeta = bsc.ObjectMeta
+	storageClass.Parameters = bsc.Parameters
+	storageClass.Provisioner = bsc.Provisioner
+	if storageClass.Labels == nil {
+		storageClass.Labels = map[string]string{}
+	}
+	storageClass.Labels[backingStorageClassLabel] = ""
+
+	if bsc.Provisioner == "" {
+		if platform != configv1.AWSPlatformType {
+			return fmt.Errorf("auto detection of provisioner is not supported for %s", platform)
+		}
+		storageClass.Provisioner = "ebs.csi.aws.com"
+	}
+
+	existingStorageClass := &storagev1.StorageClass{}
+	err := r.Client.Get(r.ctx, types.NamespacedName{Name: storageClass.Name}, existingStorageClass)
+	if errors.IsNotFound(err) {
+		// Since the StorageClass is not found, we will create a new one
+		err = r.Client.Create(r.ctx, storageClass)
+		if err != nil {
+			return err
+		}
+	} else if err != nil {
+		return err
+	} else {
+		if !reflect.DeepEqual(storageClass.Parameters, existingStorageClass.Parameters) {
+			// Since we have to update the existing StorageClass
+			// So, we will delete the existing storageclass and create a new one
+			r.Log.Info("StorageClass needs to be updated, deleting it.", "StorageClass", storageClass.Name)
+			err = r.Client.Delete(r.ctx, existingStorageClass)
+			if err != nil {
+				r.Log.Error(err, "Failed to delete StorageClass.", "StorageClass", existingStorageClass.Name)
+				return err
+			}
+			r.Log.Info("Creating StorageClass.", "StorageClass", storageClass.Name)
+			err = r.Client.Create(r.ctx, storageClass)
+			if err != nil {
+				r.Log.Info("Failed to create StorageClass.", "StorageClass", storageClass.Name)
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func getBackingStorageClasses(r *StorageClusterReconciler) (*v1.StorageClassList, error) {
+	existingBackingStorageClasses := &v1.StorageClassList{}
+	selector, err := metav1.LabelSelectorAsSelector(&metav1.LabelSelector{
+		MatchExpressions: []metav1.LabelSelectorRequirement{
+			{
+				Key:      backingStorageClassLabel,
+				Operator: metav1.LabelSelectorOpExists,
+			},
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	err = r.Client.List(r.ctx, existingBackingStorageClasses, &client.ListOptions{LabelSelector: selector})
+	if err != nil {
+		return nil, err
+	}
+
+	return existingBackingStorageClasses, nil
+}
+
+// ensureDeleted deletes the backing storageclasses
+func (obj *backingStorageClasses) ensureDeleted(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) (reconcile.Result, error) {
+	existingBackingStorageClasses, err := getBackingStorageClasses(r)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	for i := range existingBackingStorageClasses.Items {
+		storageClass := &storagev1.StorageClass{}
+		storageClass.Name = existingBackingStorageClasses.Items[i].Name
+		if err := r.Client.Delete(r.ctx, storageClass); err != nil && !errors.IsNotFound(err) {
+			r.Log.Error(err, "Unable to delete BackingStorageClass.", "Name", storageClass.Name)
+			return reconcile.Result{}, err
+		}
+	}
```

Comment:
1. No need to create an empty copy. You can just use the item from the list.
2. Please us `r.delete` util instead of `r.Client.Delete`
3. Don't stop deleting because of a single error.

```suggestion
	hasErrors := false;
	for i := range existingBackingStorageClasses.Items {
		storageClass := &existingBackingStorageClasses.Items[i]
		if err := r.delete(storageClass); err != nil && !errors.IsNotFound(err) {
			r.Log.Error(err, "Unable to delete BackingStorageClass.", "Name", storageClass.Name)
			hasErrors = true;
		}
	}
	if hasErrors {
		return reconcile.Result{}, fmt.Error("Delete failed on one or more backing storage classes"
	}
```